keras/callbacks.py

```diff
@@ -658,6 +658,12 @@ def set_model(self, model):
                     if self.write_grads:
                         grads = model.optimizer.get_gradients(model.total_loss,
                                                               weight)
+
+                        def is_indexed_slices(grad):
+                            return type(grad).__name__ == 'IndexedSlices'
+                        grads = [
+                            grad.values if is_indexed_slices(grad) else grad
+                            for grad in grads]
```

Comment:
I don't think that's really necessary here. See https://github.com/fchollet/keras/issues/7489#issuecomment-319380958 for a simpler fix.